backend/src/controllers/group/get-all-groups.ts

```diff
@@ -0,0 +1,41 @@
+import { Request, Response } from "express";
+import { Like } from "typeorm";
+
+import dataSource from "../../data-source";
+import { Group } from "../../entity/group.entity";
+
+const getAllGroups = async (req: Request, res: Response) => {
+  const groupRepository = dataSource.getRepository(Group);
+
+  try {
+    const { filterByTitle, filterByContent } = req.query;
+
+    if (filterByTitle != null || filterByContent != null) {
```

Comment:
You can do this short hand, like thos: 
```suggestion
    if (filterByTitle || filterByContent) {
```